### PR TITLE
bootstrap update for Threefold nodes

### PR DESCRIPTION
**Pull request by jvwilder**

### CODE REVIEW

File: templates/bootstrap.zero-os/actions.py [hunk 1]
```diff
@@ -83,6 +83,22 @@ def try_authorize(job, logger, netid, member, zerotier):
         member = resp.json()
     zerotier_ip = member['config']['ipAssignments'][0]
 
+    # execute hardwarecheck if necessary
+    hwc_producers=service.producers.get('hardwarecheck')
+    if hwc_producers != None:
```
Comment (zaibon) [inline templates/bootstrap.zero-os/actions.py]: for boolean condition use `is not` intead of `!=`

File: templates/bootstrap.zero-os/actions.py [hunk 2]
```diff
@@ -83,6 +83,22 @@ def try_authorize(job, logger, netid, member, zerotier):
...
+        logger.info("Hardware check necessary")
+        hwcheck_job=hwc_producers[0].getJob('check', args={'ipaddr': zerotier_ip,
```
Comment (FastGeert) [inline templates/bootstrap.zero-os/actions.py]: If get producers returns a list, we should loop them

File: templates/bootstrap.zero-os/schema.capnp [hunk 3]
```diff
@@ -3,7 +3,9 @@
 struct Schema {
     zerotierNetID @0 :Text;
     zerotierToken @1 :Text;
+    # networks the new node needs to consume
```
Comment (zaibon) [inline templates/bootstrap.zero-os/schema.capnp]: Need to update the schema ID when modifying it's content.

File: templates/bootstrap.zero-os/actions.py [hunk 4]
```diff
@@ -83,6 +83,22 @@ def try_authorize(job, logger, netid, member, zerotier):
...
+                                                       'node_id': member['nodeId'],
+                                                       'jwt': get_jwt_token(service.aysrepo)})
+        j.tools.async.wrappers.sync(hwcheck_job.execute())
+
+    # execute odoo registration if necessary
+    odoo_producers=service.producers.get('odooregistration')
```
Comment (FastGeert) [inline templates/bootstrap.zero-os/actions.py]: A more generic name will be better. Eg erp_registration.

An implementation for odoo can then be: erp_registration.odoo

Comment (FastGeert) [inline templates/bootstrap.zero-os/actions.py]: Also erp registration should be moved to the complete bottom of the method.

**Pull request merged**